mlrun/datastore/datastore.py

```diff
@@ -175,6 +179,9 @@ def object(
             )
 
         store, subpath = self.get_or_create_store(url, secrets=secrets)
+        schema, endpoint, parsed_url = parse_url(url)
+        if endpoint and endpoint in url and schema == "dbfs":
+            url = url.replace(endpoint, "")
```

Comment:
If we need to drop the authority (endpoint), why can't we do it in `DBFSStore` itself?

Reply:
Why would other file systems fail if you move this to `DBFSStore`?

Reply:
The replace isn't safe either... The endpoint could be repeated in the path, and the replace would corrupt it. It's a bit of a weird scenario, but completely legal. E.g. `dbfs://mysite.com/mysite.com`

Reply:
How about overriding `def url(self)` in `DBFSStore`? With a safer implementation.